### ISSUE TITLE
pauseResume

### ISSUE THREAD

**Issue by podaen** [schreibfaul1/ESP32-audioI2S#368]
The name of the function is confusion. It's returning a boolean but only if it is set to pauze. I would advise you to name it AUDIO::pauze instead.

I suspect the player set the m_f_running automatically to false when it stops playing and true when it starts playing?

Kind regards
Dave

**Comment by podaen** [comment]
The thing was that there is no function to only pause it or resume it. But with combining two it can be done.

```
void Pause() {
	if (audio.isRunning()) {
		audio.pauseResume();
	}
}
void Resume() {
	if (!audio.isRunning()) {
		audio.pauseResume();
	}
}
```

I search for this because the scratching sound is less when changing a station by pause it. But the loop is so fast that the decoder is not yet empty before I start a new host and I think that's where the scratching sound is coming from. Can I check if the decoder is empty?

**Comment by podaen** [comment]
Oh so this is stopping the song entirely audio.stopSong(). But here is the think. I have streams that are sending huge packages that flows in the decoder. If I do this  playI2Sremains(); it will run for quit some time I think. Can I just stop the output to decoder when a frame is complete and then empty the decoder?

**Comment by podaen** [comment]
Solved it like commented in #372
